storage/innobase/row/row0ins.cc

```diff
@@ -3393,7 +3400,8 @@ row_ins_index_entry(
 		if (auto t= trx->check_bulk_buffer(index->table)) {
 			/* MDEV-25036 FIXME: check also foreign key
 			constraints */
-			ut_ad(!trx->check_foreigns);
+			ut_ad(index->table->skip_alter_undo
+			      || !trx->check_foreigns);
```

Comment:
It would be good to have a comment that explains where the `FOREIGN KEY` constraints will be enforced during `ALTER TABLE…ALGORITHM=COPY`.

Reply:
I think that a source code comment here about the `FOREIGN KEY` constraint enforcement would be useful. In which function is that check executed?

Reply:
Unfortunately, it looks like this comment was never addressed, and we introduced the regression [MDEV-34756](https://jira.mariadb.org/browse/MDEV-34756) as a result.

Reply:
There was a misunderstanding that caused a regression that will be fixed in #3464. I think that from now on, it is better to always explain how a request was resolved. I could also have been more explicit in requesting that the `FOREIGN KEY` enforcement code path be covered in the regression test suite.